packages/react-components/src/components/TagInput/Tag.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import cx from 'classnames';
+import { Text } from '../Text';
+import { Icon, IconSizeName } from '../Icon';
+import { Close } from '@livechat/design-system-icons/dist/material';
+import { EditableTagContent } from './EditableTagContent';
+
+const baseClass = 'lc-tag-input__tag';
+
+export interface ITagProps extends React.HTMLAttributes<HTMLDivElement> {
+  inputRef: React.RefObject<HTMLInputElement>;
+  index: number;
+  update: (idx: number, value: string) => void;
+  remove: (idx: number) => void;
+  validator?: (val: string) => boolean;
+  children: string;
+}
+
+export const Tag: React.FC<ITagProps> = ({
+  className = '',
+  children,
+  index,
+  remove,
+  validator,
+  inputRef,
+  update,
+  ...restProps
+}) => {
+  const valid = React.useMemo(() => {
```

Comment:
```suggestion
  const isValid = React.useMemo(() => {
```

Reply:
anyway, do You think it's worth to memoize it?

Reply:
@mreszke I made it memoized because of the validator function, it will be passed from component props and can be basically anything, so I thought it would be better to memorize it 😄

Reply:
@przemuh wdyt?

Reply:
Discussed offline :)